src/AsyncWebSocket.cpp

```diff
@@ -1309,7 +1298,8 @@ AsyncWebSocketResponse::AsyncWebSocketResponse(const String& key, AsyncWebSocket
 #ifdef ESP8266
     sha1(key + WS_STR_UUID, hash);
 #else
-    String k = key + WS_STR_UUID;
+    String k(key);
+    k += WS_STR_UUID;
```

Comment:
Can you please change to that to avoid reallocating and internal buffer copy:
```c++
#define WS_STR_UUID_LEN 36

[...]

String k;
k.reserve(key.length() + WS_STR_UUID_LEN)
k += key;
k += WS_STR_UUID;
```

Reply:
If we are preserved the target string the better way to concatenate is:
```cpp
k.concat(key);
k.concat(WS_STR_UUID);
```
this way is more memory allocation friendly than ordinary `k += ...` IMHO...

Reply:
@Pablo2048 : please correct me if I'm wrong, but `+=` of `String` is implemented as:
```c++
        String & operator +=(const char *cstr) {
            concat(cstr);
            return (*this);
        }
```
So it is doing a concat and returns the same reference. I think the implementors made it in a way to avoid a copy.

Reply:
:D well you got me. I didn't expect that the += operator is overloaded. You are right, in this use it's the same in view of memory allocation...